datasets/turkish_treebank/turkish_treebank.py

```diff
@@ -0,0 +1,123 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+import conllu
+
+import datasets
+
+
+_CITATION = """\
+@inproceedings{kayadelen-ozturel-bohnet:2020:LREC,
+  author = {Kayadelen, Tolga  and  \"{O}zt\"{u}rel, Adnan  and  Bohnet, Bernd},
+  title = {A Gold Standard Dependency Treebank for Turkish},
+  booktitle = {Proceedings of The 12th Language Resources and Evaluation
+    Conference},
+  month = {May},
+  year = {2020},
+  address = {Marseille, France},
+  publisher = {European Language Resources Association},
+  pages = {5158--5165},
+  url = {https://www.aclweb.org/anthology/2020.lrec-1.634}
+}
+"""
+
+_DESCRIPTION = """\
+Turkish Web Treebank (TWT, ISLRN: 177-333-742-633-6) consists of 4,851 sentences (66,466 words, and 81,370 inflectional groups), which are manually annotated for segmentation, morphology, part-of-speech and dependency relations. It is composed of two sections: web and Wikipedia. Web section is built by sampling and annotating 2,541 sentences from a representative set of Turkish Forum, Blog, How-to, Review & Guides webpages. Wikipedia section is built by sampling a sentence from 2,310 Turkish Wikipedia pages and annotating them.
+"""
+
+_HOMEPAGE = "https://github.com/google-research-datasets/turkish-treebanks"
+
+_WEB_URL = "https://raw.githubusercontent.com/google-research-datasets/turkish-treebanks/master/data/web.conllu"
+_WIKI_URL = "https://raw.githubusercontent.com/google-research-datasets/turkish-treebanks/master/data/wiki.conllu"
+_VERSION = "1.0.0"
+_SOURCES = ["wiki", "web"]
+_ALL = "all"
+
+
+class TurkishTreebankConfig(datasets.BuilderConfig):
+    def __init__(self, *args, sources=None, **kwargs):
+        super().__init__(*args, version=datasets.Version(_VERSION, ""), **kwargs)
+        self.sources = sources
+
+
+class TurkishTreebank(datasets.GeneratorBasedBuilder):
+    BUILDER_CONFIGS = [
+        TurkishTreebankConfig(name=source, sources=[source], description=f"Source: {source}.") for source in _SOURCES
+    ] + [
+        TurkishTreebankConfig(
+            name=_ALL,
+            sources=_SOURCES,
+            description=f"All sources included: wiki, web",
+        )
+    ]
+    BUILDER_CONFIG_CLASS = TurkishTreebankConfig
+    DEFAULT_CONFIG_NAME = _ALL
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "id": datasets.Value("int32"),
+                    "token_ids": datasets.Sequence(datasets.Value("int32")),
+                    "tokens": datasets.Sequence(datasets.Value("string")),
+                    "lemmas": datasets.Sequence(datasets.Value("string")),
+                    "upos_tags": datasets.Sequence(datasets.Value("string")),
+                    "xpos_tags": datasets.Sequence(datasets.Value("string")),
```

Comment:
can you use ClassLabel here ?
you can find an example in [conll2003](https://github.com/huggingface/datasets/blob/master/datasets/conll2003/conll2003.py)

Reply:
(up)

What do you think about using Classlabel here ?

sorry for the spam ^^